spec/api/user_spec.rb

```diff
@@ -325,5 +325,314 @@ def test_unicode_data(text)
+          thread2 = make_thread(@user1, "Other thread", DFLT_COURSE_ID, "Thread 2", :question)
+          comment1 = make_comment(@user1, thread1, "Comment1-1")
+          comment2 = make_comment(@user1, thread2, "Comment2-1")
+          comment3 = make_comment(@user1, thread2, "Comment2-2")
+          reply1 = make_comment(@user1, comment1, "Reply1-1-1")
+          reply2 = make_comment(@user1, comment2, "Reply1-2-1")
+
+          check_social_stats(make_request(@user1.id, DFLT_COURSE_ID, nil), {@user1.id => make_social_stats(2,3,2,0,0,0,5,0)})
+          check_social_stats(make_request(@user1.id, DFLT_COURSE_ID, nil, :discussion), {@user1.id => make_social_stats(1,1,1,0,0,0,2,0)})
+          check_social_stats(make_request(@user1.id, DFLT_COURSE_ID, nil, :question), {@user1.id => make_social_stats(1,2,1,0,0,0,3,0)})
+        end
+      end
+
+      describe "all users" do
+        before :each do
+          @thread1 = make_thread(@user1, "Some thread", DFLT_COURSE_ID, "Thread 1", :discussion)
+          @thread2 = make_thread(@user2, "Other thread", DFLT_COURSE_ID, "Thread 2", :discussion)
+          @thread3 = make_thread(@user2, "Other thread", DFLT_COURSE_ID, "Thread 2", :question)
+          @comment1 = make_comment(@user2, @thread1, "Comment1-1")
+          @comment2 = make_comment(@user2, @thread1, "Comment1-2")
+          @comment3 = make_comment(@user2, @thread2, "Comment2-1")
+          @comment4 = make_comment(@user1, @thread3, "Comment3-1")
+          @reply1 = make_comment(@user1, @comment1, "Reply1-1-1")
+          @reply2 = make_comment(@user2, @comment1, "Reply1-1-2")
+          @reply3 = make_comment(@user1, @comment2, "Reply1-2-1")
+
+          set_votes(@user1, "up", [@thread2, @comment2, @comment3])
+          set_votes(@user2, "up", [@thread1, @reply1])
+          set_votes(@user3, "up", [@thread1, @thread2, @thread3])
+
+          set_flags(@user1, [@comment1, @comment2, @comment4])
+          set_flags(@user2, [@thread1, @reply1])
+          set_flags(@user3, [@thread1, @thread2])
+
+          subscribe(@thread1, [@user3])
+          subscribe(@thread2, [@user1, @user3])
+          subscribe(@thread3, [@user1])
+        end
+
+        it "returns correct stats for all users" do
+          check_social_stats(make_request('*', DFLT_COURSE_ID), {
+            @user1.id => make_social_stats(1,1,2,3,0,4,5,1),
+            @user2.id => make_social_stats(2,3,1,5,0,3,2,3),
+          })
+        end
+
+        it "filters by end_date" do
+          [@thread1, @comment1, @reply1].each do |content|
+            content.created_at = DateTime.new(2015, 02, 28)
+            content.save!
+          end
+
+          [@thread2, @comment2, @reply2].each do |content|
+            content.created_at = DateTime.new(2015, 03, 12)
+            content.save!
+          end
+
+          [@thread3, @comment3, @reply3].each do |content|
+            content.created_at = DateTime.new(2015, 03, 15)
+            content.save!
+          end
```

Comment:
Should also set `created_at` for `@comment4` somewhere as otherwise I guess it defaults to current date and is going to start giving different results after March 15th.